feat: Add tables search box

erezrokah opened the pull request:
<!-- 🎉 Thank you for making CloudQuery awesome by submitting a PR 🎉 -->

#### Summary

Fixes https://github.com/cloudquery/cloudquery/issues/8154.

The PR uses `getStaticProps` and `getStaticPaths` (the latter with an empty paths result) to load all tables on demand.
Once the table link is accessed it is generated and cached forever.

Based on https://github.com/shuding/nextra/blob/2f7f5516a66782ac3d6a02d427a34075504f4151/examples/swr-site/pages/remote/%5Bslug%5D.mdx?plain=1#L2.

Interesting bits are:
 -  `tables.js` to create the search data 
 - `Search.tsx` search component
 - `Makefile` changes to put the files in the right place and normalize the links to match the Website sturcture
 - `[table].mdx` files to generate the content on demand

<!--
Use the following steps to ensure your PR is ready to be reviewed

- [ ] Read the [contribution guidelines](../blob/main/CONTRIBUTING.md) 🧑‍🎓
- [ ] Run `make lint` to ensure the proposed changes follow the coding style 🚨 (install golangci-lint [here](https://golangci-lint.run/usage/install/#local-installation))
- [ ] Run `make test` to ensure the proposed changes pass the tests 🧪
- [ ] If changing a source plugin run `make gen` to ensure docs are up to date 📝
- [ ] Ensure the status checks below are successful ✅
--->

candiduslynx commented on website/pages/docs/plugins/sources/alicloud/tables.md:
```diff
@@ -2,8 +2,8 @@
-- [alicloud_oss_buckets](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/alicloud/docs/tables/alicloud_oss_buckets.md)
-  - [alicloud_oss_bucket_stats](https://github.com/cloudquery/cloudquery/blob/main/plugins/source/alicloud/docs/tables/alicloud_oss_bucket_stats.md)
\ No newline at end of file
+- [alicloud_bss_bill](/docs/plugins/sources/alicloud/tables/alicloud_bss_bill)
```
I wonder if it should be absolute path (should the path be prefixed with `/`)?

erezrokah replied:
Good point, I made those relative

candiduslynx commented on website/pages/docs/plugins/sources/gcp/tables/[table].mdx:
```diff
@@ -0,0 +1,25 @@
+{/* <!-- vale off --> */}
```
I think that this code should somehow be extracted and called only with `gcp` param (as the table gen is repeated across all source plugins).

erezrokah replied:
I think https://github.com/cloudquery/cloudquery/pull/8672/commits/e63ff01d45778cf8287ba91c04e4563fbe3e3ab4 is the best we can do

yevgenypats commented:
Looks great! Do we want to do the same as we did with the docs and have the tables now only on our website, to save space and duplication workflows (we can do this also in follow-up)?

erezrokah commented:
The `Summarize changes` workflow will be fixed posted merge (as it runs on `pull_request_target). See fix in https://github.com/cloudquery/cloudquery/pull/8672/files#diff-79fc9a700730d97b802a03cdd894962c175ff101b1f836c486eb66a9d746ec47

erezrokah commented:
> Looks great! Do we want to do the same as we did with the docs and have the tables now only on our website, to save space and duplication workflows (we can do this also in follow-up)?

Great point, I removed the tables from GitHub and changed the GitHub `README.md` files to point to our Website
